From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Mullin <tim@cpanel.net>
Date: Mon, 3 Aug 2020 10:46:46 -0500
Subject: [PATCH] Override RPATH for zip target

Case EA-9181:

CMake calculates the build rpath for the zip target based
on the libraries to which it links.  For some reason, it is
ending the build rpath with a colon.  To prevent this we
must override the rpath for the zip target with one we
explicitly provide.
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d00a4f9..a008157 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -311,6 +311,8 @@ if(LIBZIP_DO_INSTALL)
   install(FILES ${PROJECT_BINARY_DIR}/libzip.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
 
+set_target_properties(zip PROPERTIES BUILD_WITH_INSTALL_RPATH TRUE INSTALL_RPATH $ENV{CPANEL_LIBZIP_RPATH})
+
 # fixed size integral types
 
 if(HAVE_INTTYPES_H_LIBZIP)
